File: src/main/java/com/linux/auth/control/AuthenticationControl.java
package com.linux.auth.control;

import com.vaadin.flow.server.VaadinServletRequest;
import io.quarkus.oidc.*;
import io.quarkus.security.identity.SecurityIdentity;
import org.eclipse.microprofile.jwt.JsonWebToken;
import org.jboss.logging.Logger;

import javax.annotation.PostConstruct;
import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;

@ApplicationScoped
public class AuthenticationControl {

    private static final Logger l = Logger.getLogger(AuthenticationControl.class);

    @Inject
    @IdToken
    JsonWebToken jwt;

    @Inject
    SecurityIdentity securityIdentity;

    @Inject
    OidcSession oidcSession;


    @PostConstruct
    void init() {
        l.infov("------idToken name : {0} ------", jwt);
        l.infov("------ securityIdentity.roles : {0} ------", securityIdentity.getRoles());
        l.infov("------ VaadinServletRequest.getCurrent().userPrincipal : {0} ------", VaadinServletRequest.getCurrent().getUserPrincipal());
    }

    public String email() {
        return jwt.getName();
    }

    public String givenName() {
        return jwt.getClaim("given_name");
    }

    public String familyName(){
        return jwt.getClaim("family_name");
    }

    public void logout() {
        oidcSession.logout().await().indefinitely();
    }
}

File: src/main/kotlin/com/linux/auth/control/AuthenticationControl.kt
package com.linux.auth.control

import com.vaadin.flow.server.VaadinServletRequest
import io.quarkus.oidc.IdToken
import io.quarkus.security.identity.SecurityIdentity
import org.eclipse.microprofile.jwt.JsonWebToken
import org.jboss.logging.Logger
import javax.annotation.PostConstruct
import javax.enterprise.context.RequestScoped
import javax.inject.Inject

@RequestScoped
class AuthenticationControl(@Inject @IdToken var jwt: JsonWebToken, @Inject var securityIdentity: SecurityIdentity) {
    private val l = Logger.getLogger(AuthenticationControl::class.java)

    @PostConstruct
    fun init() {
        l.infov("------idToken name : $jwt ------")
        l.infov("------ securityIdentity.roles : ${securityIdentity.roles} ------")
        l.info("------ VaadinServletRequest.getCurrent().userPrincipal : ${VaadinServletRequest.getCurrent().userPrincipal} ------")
    }

    fun email(): String = jwt.name

    fun givenName(): String = jwt.getClaim("given_name")

    fun familyName(): String = jwt.getClaim("family_name")

}